app/uk/gov/hmrc/nationalinsurancerecord/controllers/NationalInsuranceRecordController.scala

```diff
@@ -38,7 +38,7 @@ trait NationalInsuranceRecordController extends BaseController with HeaderValida
       content,
       selfLink,
       Some(Vector("taxYears" -> years.map(taxYear => halResourceSelfLink(Json.toJson(taxYear),
-        nationalInsuranceTaxYearHref(nino, taxYear.taxYear))).toVector))
+        nationalInsuranceTaxYearHref(nino, TaxYear(taxYear.taxYear)))).toVector))
```

Comment:
Below controllers getTaxYear method is taking TaxYear object and I might need to change entire chain to accept String.? 
createLink(uk.gov.hmrc.nationalinsurancerecord.controllers.live.routes.NationalInsuranceRecordController.getTaxYear(nino,taxYear).url)